runtime/common/src/crowdloan/migration.rs

```diff
@@ -29,16 +29,28 @@ pub mod crowdloan_index_migration {
 	pub fn pre_migrate<T: Config>() -> Result<(), &'static str> {
 		// `NextTrieIndex` should have a value.
 		generate_storage_alias!(Crowdloan, NextTrieIndex => Value<FundIndex>);
-		let next_index = NextTrieIndex::take().unwrap_or_default();
+		let next_index = NextTrieIndex::get().unwrap_or_default();
```

Comment:
I assume this fixed the issue with the post-migrate check `next_index > 0`? If so that is pretty subtle; good to keep in mind that we should advise avoiding mutable storage ops in try-runtime checks cc @kianenigma

Reply:
yup, and subtle indeed.

Reply:
A good idea would be actually to get the storage root before `pre_migrate` is run, and check it is the same after `pre_migrate` is run, or panic.

Reply:
You can do it if it makes sense for your code, but we can't always assume it is the case: I imagine scenarios where you want pre-migrate and post-migrate to "talk to each other". In this case, my workaround would be to write some value to a "fake" storage key in `pre_migrate`, and read in `post_migrate`. 

For example, you want to make sure that a migrations that alters the format of some storage item, does not alter its count. In this case you want to write the previous-count in pre-migrate somewhere, and read it back in post-migrate. 

There are even some utilities made for this: https://paritytech.github.io/substrate/master/frame_support/traits/trait.OnRuntimeUpgradeHelpersExt.html#method.get_temp_storage

Reply:
Resolving this with the follow actions:
- an issue to implement data passing between hooks without modifying runtime storage: https://github.com/paritytech/substrate/issues/10824
- small doc update to warn users of mutable storage ops in hooks: https://github.com/paritytech/substrate/pull/10822